### ISSUE TITLE
Support the `document` core type

### ISSUE THREAD

**Issue by jobarr-amzn** [amzn/ion-schema-rust#123]
`ion-schema-rust` does not seem to support the `document` core type, described in the Ion schema spec [here](https://amzn.github.io/ion-schema/docs/isl-1-0/spec#core-types).

Given this content adapted from the [`document` example](https://amzn.github.io/ion-schema/docs/isl-1-0/spec#document) in the Ion Schema specification docs:
```plain
❯ cat document_example.isl
$ion_schema_1_0
```
```ion
schema_header::{
}

type::{
  name: IonSchema,
  type: document,
  ordered_elements: [
    { type: Header, occurs: optional },
    { type: Type,   occurs: range::[0, max] },
    { type: Footer, occurs: optional },
  ],
}

type::{
  name: Header,
  type: struct,
  annotations: [required::schema_header],
  fields: {
    imports: ImportList,
  },
}

type::{
  name: ImportList,
  type: list
}

type::{
  name: Type,
  type: struct,
  annotations: [required::type],
  fields: {
    name: symbol,
  },
}

type::{
  name: Footer,
  type: struct,
  annotations: [required::schema_footer],
}

schema_footer::{
}
```

`ion-schema-rust` fails because there is no known `document` type:

```plain
❯ ion beta schema validate \
--input document_example.isl \
--schema document_example.isl \
--directory . \
--type IonSchema
```
```plain
Error: Could not resolve type reference: "document" does not exist
```

**Comment by desaikd** [comment]
`ion-schema-rust` does support `document core type. I tried reproducing this issue to see if there was any other bug here. It seems to work fine withis repository's example CLI. 
I tried the following command:
```
cargo run --package ion-schema --example schema load --directory <AUTHORITY-FOLDER> --schema <SCHEMA-FILE>
```
Output for the above command gener4ates a `Schema` struct without any errors. 

How was this issue produced? If you have used `ion-cli` then it is pointing to an earlier version of `ion-schema-rust` and hence this issue might be produced from there.
I will be updating that to the latest version after a new release with latest GitHub changes for `ion-schema-rust`.
